### ISSUE TITLE
helm test exits 0 under error conditions

### ISSUE THREAD

**Issue by ahawkins** [kubernetes/helm#2313]
Helm version `2.2.3`

```
kviberg $ helm test test-efritin
RUNNING: test-efritin-kviberg-efr-smoke-test
ERROR: pods "test-efritin-kviberg-efr-smoke-test" already exists
kviberg $ echo $?
0
```

`helm test` should exit non-zero if there is an error running the test.

**Comment by xmik** [comment]
same issues: #2166, #2275

**Comment by technosophos** [comment]
Closed in lieu of #2166, which is already on the roadmap.
